### ISSUE TITLE
Compressing with dictionaries seems broken

### ISSUE THREAD

**Issue by kontura** [zchunk/zchunk#43]
Our CI for `createrepo_c` started failing with the latest `zchunk-1.1.14-1` release.

To reproduce it seems sufficient to do:
```
$ echo "primary foobar" > dictionary.zdict
$ echo "primary primary primary data" > data
$ zck -D dictionary.zdict data
$ unzck data.zck
Error reading compressed dict: zstd decompression error: Corrupted block detected
```

I am not sure if such a dictionary is legal but it was working before https://github.com/zchunk/zchunk/commit/313b083a28ceb4c11286f6e625d8632db8e53eb7.

I have also tried to compress a file with a real dictionary generated by `zstd` and I got `Error reading compressed dict: zstd decompression error: Dictionary mismatch` when running `unzck` on it but it is working fine if I downgrade to `1.1.12`.

**Comment by jdieter** [comment]
This is incredibly embarrassing.  I've found that the test were validating that a previously created zchunk file could be decompressed and that a regular file could be compressed correctly, but were not checking whether the newly created zchunk files could be decompressed again. :facepalm: When I switched to the new zstd interface, I was compressing the dictionary _using the dictionary_, which had the side effect of making it impossible to decompress. :facepalm: :face_with_head_bandage:

I've fixed the bug (and added in new tests) in 813a1f66083eec4ed925c74a43198f57d973c53a, and have untagged the 1.1.14 release.
